numpy/random/generator.pyx

```diff
@@ -3857,8 +3866,12 @@ cdef class Generator:
                     buf[...] = x[j]
                     x[j] = x[i]
                     x[i] = buf
+            x = np.swapaxes(x, 0, axis)
         else:
             # Untyped path.
+            axis = normalize_axis_index(axis, np.ndim(x))
+            if axis != 0 and axis != -1:
```

Comment:
`axis != -1` is always true because `0 <= normalize_axis_index(axis, np.ndim(x)) < np.ndim(x)` is always true. See @eric-wieser's [comment above](https://github.com/numpy/numpy/pull/13829/files#r324054975)

Reply:
@wkschwartz oh, yes.. you are right.